### PR TITLE
Rebasing release-0.8.0-RC2 branch to master

### PR DESCRIPTION
**Pull request by imesh**
This pull request rebases release-0.8.0-RC2 branch to master.

### CODE REVIEW

File: src/main/java/org/ballerinalang/docgen/docs/html/HtmlDocumentWriter.java [hunk 1]
```diff
@@ -91,14 +96,33 @@ public void write(Collection<BLangPackage> packages) throws IOException {
         if (BallerinaDocUtils.isDebugEnabled()) {
             out.println("Generating HTML API documentation...");
         }
-        Files.createDirectories(Paths.get(outputFilePath));
```
Comment (nirmal070125) [inline src/main/java/org/ballerinalang/docgen/docs/html/HtmlDocumentWriter.java]: @imesh isn't this make it impossible to create a new output directory?

Comment (imesh) [inline src/main/java/org/ballerinalang/docgen/docs/html/HtmlDocumentWriter.java]: I'm sorry Nirmal I didn't get you, the above statement creates new directories for outputFilePath if they do not exist.

BTW this pull request does not include any changes, this is just to rebasing changes in release-0.8.0-RC2 branch to master. Thanks!

Comment (nirmal070125) [inline src/main/java/org/ballerinalang/docgen/docs/html/HtmlDocumentWriter.java]: @imesh hmm.... it shows as this PR removes that line?? 
screen shot 2017-02-22 at 11 02 32 am

Comment (nirmal070125) [inline src/main/java/org/ballerinalang/docgen/docs/html/HtmlDocumentWriter.java]: In fact, I see https://github.com/ballerinalang/docerina/pull/70/files has removed the same line. Hence, the question.

Comment (imesh) [inline src/main/java/org/ballerinalang/docgen/docs/html/HtmlDocumentWriter.java]: Thanks Nirmal, I'm sorry I missed that. Let me fix that.

Comment (nirmal070125) [inline src/main/java/org/ballerinalang/docgen/docs/html/HtmlDocumentWriter.java]: No worries Imesh! Thanks.

**Comment by imesh** [comment]
Closing this pull request, will apply release branch changes by cherry picking the commits.

**Pull request closed without merging**